rskj-core/src/main/java/co/rsk/Start.java

```diff
@@ -163,6 +167,7 @@ public void startNode(String[] args) throws Exception {
             enablePeerDiscovery();
         }
 
+        messageHandler.start();
```

Comment:
TL;DR;: I think the order is wrong and this should be started before `rsk.init()`.

In the old code, `@Bean public Rsk getRsk` depended on `NodeMessageHandler`. This means that `@Bean public NodeMessageHandler getNodeMessageHandler` was executed before `getRsk`. Which means that `messageHandler.start();` was executed before `rsk.init()`.
I don't necessarily think this is wrong, but let's try to maintain the same behavior.

Reply:
great catch, thanks!